brick-editor.js

```diff
@@ -240,11 +238,36 @@ function onRangeDelete() {
     var ast = attemptParse(buffer);
     var selection = getSelected();
     if (ast && selection) {
-        selectionBranch(ast, selection);
+        selectionBranch(buffer, ast, selection);
     }
     updateEditorState();
 }
 
+/** 
+ * Set value of editor using executeEdits to preserve undo stack
+ *
+ * @param {string} oldBuffer - String being replaced. 
+ * @param {string} newBuffer - String replacing oldBuffer.
+ * @returns {undefined}
+*/
+function setValue(oldBuffer, newBuffer) {
+    // get range of editor model 
+    var range = editor.getModel().getFullModelRange();
+    // call execute edits on the editor 
+    editor.executeEdits(oldBuffer, [{ identifier: "insert", range: range, text: newBuffer }]);
```

Comment:
Do we really need `oldBuffer` as a parameter, or is it always `editor.getValue()`?